server.py

```diff
@@ -146,6 +143,20 @@ def dashboard():
     contests = Contest.select().order_by(Contest.start_time)
     return bottle.template("dashboard.html", contests=contests)
 
+@app.get("/stats")
+@login_required
+def statistics():
+    sub_history_temp = Submission.select(Contest.code, ContestProblems.question, Submission.time, Submission.is_correct)\
+        .where(Submission.user == Session.get(Session.token == bottle.request.get_cookie("s_id")).user)\
+        .join(ContestProblems, on=(Submission.contestProblem == ContestProblems.id)) \
+        .join(Session, on=(Submission.user == Session.user))\
+        .switch() \
+        .join(Contest, on=(ContestProblems.contest == Contest.id)) \
+        .order_by(Submission.time.desc())
+    sub_history = list(sub_history_temp.tuples())
```

Comment:
can we get rid of the intermediate temp var? Just put it directly in the expected variable?

Reply:
also, since we are using this later on in both template and the code, it might be a good idea to do `dicts()` instead of `tuples()`. Makes the later code clearer:

```python
sub_stats_correct = len([sub_history for sub in sub_history if sub['is_correct']])
```